src/rules/noStringLiteralRule.ts

```diff
@@ -19,34 +19,36 @@
 export class Rule extends Lint.Rules.AbstractRule {
     public static FAILURE_STRING = "object access via string literals is disallowed";
 
-    public apply(syntaxTree: TypeScript.SyntaxTree): Lint.RuleFailure[] {
-        return this.applyWithWalker(new NoStringLiteralWalker(syntaxTree, this.getOptions()));
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new NoStringLiteralWalker(sourceFile, this.getOptions()));
     }
   }
 
 class NoStringLiteralWalker extends Lint.RuleWalker {
-    public visitElementAccessExpression(node: TypeScript.ElementAccessExpressionSyntax): void {
-        this.handleElementAccessExpression(node);
-        super.visitElementAccessExpression(node);
-    }
-
-    private handleElementAccessExpression(node: TypeScript.ElementAccessExpressionSyntax) {
+    public visitElementAccessExpression(node: ts.ElementAccessExpression) {
         var argument = node.argumentExpression;
-        var id = TypeScript.fullText(argument);
+        var accessor = argument.getText();
```

Comment:
oh, i see why you're calling it accessor. accessorText is more appropriate here.